internal/service/vmservice/vm.go

```diff
@@ -225,6 +239,20 @@ func reconcileVirtualMachineConfig(ctx context.Context, machineScope *scope.Mach
 		}
 	}
 
+	// custom tags
+	if machineScope.ProxmoxMachine.Spec.Tags != nil {
+		tags := strings.Split(machineScope.VirtualMachine.VirtualMachineConfig.Tags, ";")
+		length := len(tags)
+		for _, tag := range machineScope.ProxmoxMachine.Spec.Tags {
+			if !machineScope.VirtualMachine.HasTag(tag) {
+				tags = append(tags, tag)
+			}
+		}
+		if len(tags) > length {
```

Comment:
this check does not account for removal of tags. is this intended?

Reply:
Yes, so we don't delete existing tags. 
The IP tag i mean.
Which means tags are also immutable as per the machine.

Reply:
then you need to add this to the webhook or a CRD kubebuilder annotation